src/parser.ts

```diff
@@ -531,13 +530,23 @@ export class Parser {
       }
 
       const parent = getParentType(prop);
+      const declarations = prop.declarations || [];
+      const baseProp = baseProps.find((p) => p.getName() === propName);
+
+      const required =
+        !isOptional(prop) &&
+        !hasCodeBasedDefault &&
+        // If in a intersection or union check original declaration for "?"
+        // @ts-ignore
+        declarations.every((d) => !d.questionToken) &&
```

Comment:
One caveat of the new above approach is that the `flags` for each prop mark it as `required` for some reason. To fix this I look at the declarations of the properties (it could be on multiple interfaces) to see if it was marked as optional.

Then on the line below I also check that if there is a property returned from `propsType.getProperties();` check if that property is optional. This fixed the test case where we extend from a picked interface.